src/snowflake/snowpark/types.py

```diff
@@ -11,8 +11,15 @@
 from typing import Generic, List, Optional, Type, TypeVar, Union
 
 import snowflake.snowpark._internal.analyzer.expression as expression
-from snowflake.connector.options import installed_pandas, pandas
-from snowflake.snowpark._internal.utils import quote_name
+import snowflake.snowpark._internal.proto.ast_pb2 as proto
+
+# Use correct version from here:
+from snowflake.snowpark._internal.utils import installed_pandas, pandas, quote_name
+
+# TODO: connector installed_pandas is broken. If pyarrow is not installed, but pandas is this function returns the wrong answer.
+# The core issue is that in the connector detection of both pandas/arrow are mixed, which is wrong.
+# from snowflake.connector.options import installed_pandas, pandas
```

Comment:
this is a good observation. I'll create a JIRA to replace `from snowflake.connector.options import installed_pandas` -> `from snowflake.snowpark._internal.utils import installed_pandas`